donut/modules/groups/helpers.py

```diff
@@ -82,3 +82,28 @@ def get_group_data(group_id, fields=None):
     # Return the row in the form of a dict
     result = {f: t for f, t in zip(fields, result)}
     return result
+
+
+
+def get_position_data(fields=None):
+    all_returnable_fields = ["group_id", "pos_id", "pos_name"]
+    default_fields = ["user_id", "pos_id", "pos_name", "group_id"]
+
+    if fields is None:
+        fields = default_fields
+    else:
+        if any(f not in all_returnable_fields for f in fields):
+            return "Invalid field"
+
+    #s= sqlalchemy.sql.select(fields).select_from(sqlalchemy.text("positions JOIN position_holders WHERE positions.pos_id = position_holders.pos_id"))
+    
+    
+    result = flask.g.db.execute("SELECT user_id, positions.pos_id, pos_name, positions.group_id \nFROM positions JOIN position_holders WHERE positions.pos_id = position_holders.pos_id;")    
+    
+    if result is None:
+        return {}
+
+    user_position_arr = [{f: t for f,t in zip(fields, row)} for row in result]
```

Comment:
Here, if the fields you request are anything other than default_fields, you may return incorrect results since it would zip up all of the fields specified in the argument, since the query grabs all of the default_fields from the database.

Also, I'm not sure if you want to return user_id (since user_id is not in all_returnable_fields). Look into it.